apps/redis.go

```diff
@@ -37,18 +41,31 @@ func (r MetricsReceiverRedis) Type() string {
 }
 
 func (r MetricsReceiverRedis) Pipelines() []otel.Pipeline {
-
 	if r.Address == "" {
 		r.Address = defaultRedisEndpoint
 	}
 
+	var insecure string
+	if r.CertFilePath != "" || r.KeyFilePath != "" || r.CaFilePath != "" {
+		insecure = "false"
```

Comment:
`CaFilePath == ""` doesn't imply `insecure=="false"` because you can use the system default trust store to validate the server certificate.

Reply:
I think what you meant to say is `CaFilePath == ""` doesn't imply `insecure=="true"` because that means don't use tls and is currently how it's set up. Do what we want to do then is just set it to false if nothing is set so TLS IS enabled?

Reply:
I'm not sure what you mean, so let me try to just enumerate the possible states an HTTP client socket can be in:

- HTTP (no SSL/TLS) - I think this is what `insecure: true` means
- HTTPS with server certificate validated with system trust store - `insecure: false` with no other settings
- HTTPS with server certificate validated with a custom certificate authority - `insecure: false` and `ca_file_path: /foo/bar`
- HTTPS with no server certificate validation - `insecure: false` with `insecure_skip_verify: true`

and then the last three repeated with the addition of a client certificate with `cert_file_path` and `key_file_path` specified.

All 7 of those are different states and it should be possible to configure all of them. And the default behavior should either be the first one or the second one depending on whether the server defaults to using SSL or not using SSL.

Reply:
ah ok, appreciate that, and from [what I'm seeing](https://redis.io/topics/encryption#:~:text=By%20default%2C%20Redis%20uses%20mutual,ca%2Dcert%2Ddir%20\).), Redis does not use TLS/SSL by default so the default should be `insecure: true`.

Reply:
Reached agreement offline. See the common util shared by https://github.com/GoogleCloudPlatform/ops-agent/pull/320 for the expected implementation.